### PR TITLE
Add support for multiple -f flags in snow sql

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Support executing multiple files in `snow sql` command by passing multiple `-f` flags.

### CODE REVIEW

File: src/snowflake/cli/api/utils/cursor.py [hunk 1]
```diff
@@ -18,3 +18,9 @@ def find_first_row(
 ) -> Optional[dict]:
     """Returns the first row that matches the predicate, or None."""
     return next(_rows_generator(cursor, predicate), None)
+
+
+def join_cursors(cursors: List[Iterable[SnowflakeCursor]]) -> Iterable[SnowflakeCursor]:
+    while cursors:
+        cur = cursors.pop(0)
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/api/utils/cursor.py:25]: nit: list needs to continually be modified, `for cur in cursors:` feels simpler and more performant given `cursors` is a `List`

File: src/snowflake/cli/plugins/sql/manager.py [hunk 2]
```diff
@@ -35,9 +35,30 @@ def execute(
 
         if std_in:
             query = sys.stdin.read()
-        elif file:
-            query = SecurePath(file).read_text(file_size_limit_mb=UNLIMITED)
+        elif files and len(files) == 1:
+            # If there's only one file we need to handle case of single query
+            query = SecurePath(files[0]).read_text(file_size_limit_mb=UNLIMITED)
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/plugins/sql/manager.py:40]: Just curious, why a special case for one file?

Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/plugins/sql/manager.py:40]: Non-blocking: I'd suggest removing the special case for a single file and instead tally all queries you ever found to figure out if there is a "single statement" or not, but maybe you are working around some property of `split_statements` (does it erroneously return non-statements if a file is empty of non-comment SQL?).

**Review by sfc-gh-cgorrie — APPROVED** [review]
NADE-owned files LGTM, just a couple thoughts. And it would be nice to see a new test to codify what exactly happens when empty files are included in the file list.

File: src/snowflake/cli/api/utils/cursor.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/api/utils/cursor.py:25]: Good point, adjusted

File: src/snowflake/cli/plugins/sql/manager.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/plugins/sql/manager.py:40]: Good point, changed

File: src/snowflake/cli/plugins/sql/manager.py [hunk 3]
```diff
@@ -3,30 +3,30 @@
...
 from jinja2 import UndefinedError
 from snowflake.cli.api.secure_path import UNLIMITED, SecurePath
 from snowflake.cli.api.sql_execution import SqlExecutionMixin
+from snowflake.cli.api.utils.cursor import join_cursors
 from snowflake.cli.api.utils.rendering import snowflake_sql_jinja_render
 from snowflake.cli.plugins.sql.snowsql_templating import transpile_snowsql_templates
 from snowflake.connector.cursor import SnowflakeCursor
 from snowflake.connector.util_text import split_statements
 
+SingleStatement = bool
```
Comment (sfc-gh-pczajka) [inline src/snowflake/cli/plugins/sql/manager.py:18]: How about
```suggestion
IsSingleStatement = bool
```

File: src/snowflake/cli/api/utils/cursor.py [hunk 4]
```diff
@@ -18,3 +18,8 @@ def find_first_row(
 ) -> Optional[dict]:
     """Returns the first row that matches the predicate, or None."""
     return next(_rows_generator(cursor, predicate), None)
+
+
+def join_cursors(cursors: List[Iterable[SnowflakeCursor]]) -> Iterable[SnowflakeCursor]:
```
Comment (sfc-gh-pczajka) [inline src/snowflake/cli/api/utils/cursor.py:23]: How about using `itertools.chain.from_iterable` instead?
https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable

**Pull request merged**